internal/generate/olm-catalog/descriptor/descriptor.go

```diff
@@ -115,7 +118,7 @@ func GetCRDDescriptionForGVK(apisDir string, gvk schema.GroupVersionKind) (olmap
 		crdDesc.DisplayName = kindDescriptors.displayName
 	}
 	if len(kindDescriptors.resources) != 0 {
-		crdDesc.Resources = sortResources(kindDescriptors.resources)
+		crdDesc.Resources = sortResourcesLegacy(kindDescriptors.resources)
```

Comment:
what the diff between? 
Also, I think all legacy stuff should be annotated with Deprecated to make clear that we should no longer use them.

Reply:
Right now there's no `sortResources` because the new type can be generically sorted. I'll change the name back so there's no confusion.